vignettes/rebuild_cached_data.Rmd

```diff
@@ -1,128 +1,100 @@
-
-# use diffdf package: diffdf() to compare cached vs new and review/verify differences
-diffdf(cadsl, cadsl_new)
-diffdf(cadab, cadab_new)
-diffdf(cadae, cadae_new)
-diffdf(cadaette, cadaette_new)
-diffdf(cadcm, cadcm_new)
-diffdf(cadex, cadex_new)
-diffdf(cadlb, cadlb_new)
-diffdf(cadmh, cadmh_new)
-diffdf(cadpc, cadpc_new)
-diffdf(cadpp, cadpp_new)
-diffdf(cadqs, cadqs_new)
-diffdf(cadrs, cadrs_new)
-diffdf(cadtte, cadtte_new)
-diffdf(cadvs, cadvs_new)
-diffdf(cadeg, cadeg_new)
-diffdf(cadtr, cadtr_new)
-diffdf(caddv, caddv_new)
-diffdf(cadsub, cadsub_new)
-diffdf(cadhy, cadhy_new)
-diffdf(cadqlqc, cadqlqc_new)
-
-# when diff reflects changes accurately then create environment objects to save out to cache
-cadsl <- cadsl_new
-cadab <- cadab_new
-cadae <- cadae_new
-cadaette <- cadaette_new
-cadcm <- cadcm_new
-cadeg <- cadeg_new
-cadex <- cadex_new
-cadlb <- cadlb_new
-cadmh <- cadmh_new
-cadpc <- cadpc_new
-cadpp <- cadpp_new
-cadqs <- cadqs_new
-cadrs <- cadrs_new
-cadtr <- cadtr_new
-cadtte <- cadtte_new
-cadvs <- cadvs_new
-caddv <- caddv_new
-cadsub <- cadsub_new
-cadhy <- cadhy_new
-cadqlqc <- cadqlqc_new
-
-# update cache
-save(cadsl, file = "data/cadsl.RData", compress = "xz")
-save(cadab, file = "data/cadab.RData", compress = "xz")
-save(cadae, file = "data/cadae.RData", compress = "xz")
-save(cadaette, file = "data/cadaette.RData", compress = "xz")
-save(cadcm, file = "data/cadcm.RData", compress = "xz")
-save(cadeg, file = "data/cadeg.RData", compress = "xz")
-save(cadex, file = "data/cadex.RData", compress = "xz")
-save(cadlb, file = "data/cadlb.RData", compress = "xz")
-save(cadmh, file = "data/cadmh.RData", compress = "xz")
-save(cadpc, file = "data/cadpc.RData", compress = "xz")
-save(cadpp, file = "data/cadpp.RData", compress = "xz")
-save(cadqs, file = "data/cadqs.RData", compress = "xz")
-save(cadrs, file = "data/cadrs.RData", compress = "xz")
-save(cadtr, file = "data/cadtr.RData", compress = "xz")
-save(cadtte, file = "data/cadtte.RData", compress = "xz")
-save(cadvs, file = "data/cadvs.RData", compress = "xz")
-save(caddv, file = "data/caddv.RData", compress = "xz")
-save(cadsub, file = "data/cadsub.RData", compress = "xz")
-save(cadhy, file = "data/cadhy.RData", compress = "xz")
-save(cadqlqc, file = "data/cadqlqc.RData", compress = "xz")
+# Datasets must be listed after all of their dependencies
+# e.g. adsl is a dependency for all other datasets so it is listed first.
+data_nms <- c(
+  "adsl", "adae", "adaette", "adcm", "adex", "adlb", "admh", "adpc", "adpp", "adab",
+  "adqs", "adrs", "adtte", "advs", "adeg", "adtr", "addv", "adsub", "adhy", "adqlqc"
+)
+
+# Construction of dependency tree based on formals
+data_deps <- sapply(
+  data_nms,
+  function(x) {
+    dat_args <- names(formals(paste0("r", x)))
+    dat_args[dat_args %in% data_nms]
+  }
+)
+
+updated_files <- system("git diff origin/main --name-only", intern = TRUE)
```

Comment:
I am sure this is not very stable, having seen how staged.deps works. Do you have any suggestions to improve this @pawelru?

Reply:
I have already included the check of identical data and now it works (it was updating the data attribute and rewriting to file anyway). I will add a try-catch for git, so eventually we recreate the data but if no change is happening we do not rewrite the data.

Reply:
Thanks for pointing it out ;)